### PR TITLE
SNOW-652526: update release notes for v0.9.0

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-652526

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
-## 0.9.0 (Unreleased)
+## 0.9.0 (2022-08-31)
 
 ### New Features:
 - Added support for displaying source code as comments in the generated scripts when registering UDFs.
 This feature is turned on by default. To turn if off, pass the new keyword argument `source_code_display` as `False` when calling `register()` or `@udf()`.
+- Added support for `column_names` in `df.write.save_as_table()`. You can now specify the names of columns when appending new data to an existing table.
+- Added function `get_active_session()` in module `snowflake.snowpark.context` to get the current active Snowpark session.
+- Added keyword argument `overwrite` to `session.write_pandas()` to allow overwriting contents of a Snowflake table with that of a Pandas DataFrame.
+- Added `FileOperation.put_stream()` to upload local files to a stage via file stream.
```
Comment (sfc-gh-kwagner) [inline CHANGELOG.md:10]: I would have all of these start with "Added support for" so that the wording is consistent

**Review by sfc-gh-kwagner — APPROVED** [review]
Looks good other than minor comment

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+- Added support for calling table functions from `DataFrame.select()`, `DataFrame.with_column()` and `DataFrame.with_columns` which now take parameters of type `table_function.TableFunctionCall` for columns.
+- Added `TableFunctionCall.alias` to allow aliasing the names of columns that come from the output of table function joins.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:12]:
```suggestion
- Added `TableFunctionCall.alias()` and `TableFunctionCall.as_()` to allow aliasing the names of columns that come from the output of table function joins.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 
+### Bug Fixes:
+- Fixed a bug in which batch insert should not raise error when `statement_params` is not passed to the function.
+- Fixed a bug in which column names should be quoted when `create_dataframe` is called with dicts and a given schema.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:16]:
```suggestion
- Fixed a bug in which column names should be quoted when `session.create_dataframe` is called with dicts and a given schema.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+- Fxied a bug in which creation of table shall be skipped if the table already exists and is in append mode.
+- Fxied a bug that packages with underscores cannot be added when registering UDFs.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:18]:
```suggestion
- Fixed a bug that third-party packages with underscores cannot be added when registering UDFs.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
+- Added function `get_active_session()` in module `snowflake.snowpark.context` to get the current active Snowpark session.
+- Added keyword argument `overwrite` to `session.write_pandas()` to allow overwriting contents of a Snowflake table with that of a Pandas DataFrame.
+- Added `FileOperation.put_stream()` to upload local files to a stage via file stream.
+- Added support for calling table functions from `DataFrame.select()`, `DataFrame.with_column()` and `DataFrame.with_columns` which now take parameters of type `table_function.TableFunctionCall` for columns.
+- Added `TableFunctionCall.alias` to allow aliasing the names of columns that come from the output of table function joins.
 
+### Bug Fixes:
+- Fixed a bug in which batch insert should not raise error when `statement_params` is not passed to the function.
+- Fixed a bug in which column names should be quoted when `create_dataframe` is called with dicts and a given schema.
+- Fxied a bug in which creation of table shall be skipped if the table already exists and is in append mode.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:17]:
```suggestion
- Fixed a bug in which creation of table shall be skipped if the table already exists and is in append mode.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,10 +1,24 @@
...
+- Fxied a bug that packages with underscores cannot be added when registering UDFs.
+
+### Improvements:
+- Improved function `uniform` to infer the types of inputs `max_` and `_min` and cast the limits to `IntegerType` or `FloatType` correspondingly.
```
Comment (sfc-gh-jdu) [inline CHANGELOG.md:21]:
```suggestion
- Improved function `functions.uniform()` to infer the types of inputs `max_` and `min_` and cast the limits to `IntegerType` or `FloatType` correspondingly.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,10 +1,24 @@
 # Release History
-## 0.9.0 (Unreleased)
+## 0.9.0 (2022-08-31)
 
 ### New Features:
 - Added support for displaying source code as comments in the generated scripts when registering UDFs.
 This feature is turned on by default. To turn if off, pass the new keyword argument `source_code_display` as `False` when calling `register()` or `@udf()`.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:6]: if -> it

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,10 +1,24 @@
...
+- Added support for `column_names` in `df.write.save_as_table()`. You can now specify the names of columns when appending new data to an existing table.
+- Added function `get_active_session()` in module `snowflake.snowpark.context` to get the current active Snowpark session.
+- Added keyword argument `overwrite` to `session.write_pandas()` to allow overwriting contents of a Snowflake table with that of a Pandas DataFrame.
+- Added `FileOperation.put_stream()` to upload local files to a stage via file stream.
+- Added support for calling table functions from `DataFrame.select()`, `DataFrame.with_column()` and `DataFrame.with_columns` which now take parameters of type `table_function.TableFunctionCall` for columns.
+- Added `TableFunctionCall.alias` to allow aliasing the names of columns that come from the output of table function joins.
 
+### Bug Fixes:
+- Fixed a bug in which batch insert should not raise error when `statement_params` is not passed to the function.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:15]: "raise error" -> "raise an error"

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:17]: Also, I suggest changing "shall be skipped" to "is skipped".

File: CHANGELOG.md (hunk 4 above)
Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:18]: Also, change "that" to "where" or "in which".

**Review by sfc-gh-mabrennan — APPROVED** [review]
Nice job. I only had minor suggestions. Thank you.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,10 +1,24 @@
 # Release History
-## 0.9.0 (Unreleased)
+## 0.9.0 (2022-08-31)
 
 ### New Features:
 - Added support for displaying source code as comments in the generated scripts when registering UDFs.
 This feature is turned on by default. To turn if off, pass the new keyword argument `source_code_display` as `False` when calling `register()` or `@udf()`.
+- Added support for `column_names` in `df.write.save_as_table()`. You can now specify the names of columns when appending new data to an existing table.
```
Comment (sfc-gh-yixie) [inline CHANGELOG.md:7]: This is not accurate. A new param `column_order` is added. When it's "name" and `save_as_table` is in append mode, the dataframe data is appended to an existing table by matching the dataframe's column names and the table's column names. When it's "index", it works as before.

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-yixie) [reply CHANGELOG.md:17]: Is this for `copy_into_table`? Better explicitly mention the API

Comment (sfc-gh-aling) [reply CHANGELOG.md]: updated, from the commit: https://github.com/snowflakedb/snowpark-python/commit/1d61cfab560a1381cb49cdad6693f0ea613dc0a5 it's the `df.write.save_as_table` api.

@sfc-gh-sfan , is ther other api impacted?

File: CHANGELOG.md (hunk 9 above)
Comment (sfc-gh-aling) [reply CHANGELOG.md]: my fault, I read into the analyzer part /facepalm/.

I have updated to `Added keyword argument `column_order` to `df.write.save_as_table()` to specify the matching rules when inserting data into table in append mode.`

how does it look to you now?

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-aling) [reply CHANGELOG.md]: I feel "added support for" might not be the best for all cases if we're simply adding adding argument/method/functions.

But I got your point on consistency, I have tweaked a bit to make them consistent, how do you like it now?

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]
LGTM

**Pull request merged**